hydromt/raster.py

```diff
@@ -654,24 +654,42 @@ def origin(self) -> tuple[float, float]:
     def _check_dimensions(self) -> None:
         """Validate the number and order of dimensions."""
         dims = (self.y_dim, self.x_dim)
-        da = self._obj[self.vars[0]] if isinstance(self._obj, xr.Dataset) else self._obj
-        extra_dims = [dim for dim in da.dims if dim not in dims]
+
+        def _check_dims(da: xr.DataArray, dims=dims) -> bool:
+            extra_dims = [dim for dim in da.dims if dim not in dims]
+            if len(extra_dims) == 1:
+                dims = tuple(extra_dims) + dims
+            elif len(extra_dims) > 1:
+                raise ValueError("Only 2D and 3D data arrays supported.")
+            return da.dims == dims, extra_dims
+
+        extra_dims = []
+        if isinstance(self._obj, xr.DataArray):
+            check, dim0 = _check_dims(self._obj)
+            extra_dims.extend(dim0)
```

Comment:
Do we want to extend here even if we raise a value error on the next line?

Reply:
Only when no error is raised we actually use extra_dims, so I don't think this matters